? esmf.patch
? build_config/Linux.intel.default/build_rules.mk.ORIG
? build_config/Linux.pgi.default/build_rules.mk.ORIG
? scripts/available
Index: build_config/Linux.intel.default/build_rules.mk
===================================================================
RCS file: /cvsroot/baselibs/Baselibs/src/esmf/build_config/Linux.intel.default/build_rules.mk,v
retrieving revision 1.1.5.1
diff -b -u -p -r1.1.5.1 build_rules.mk
--- build_config/Linux.intel.default/build_rules.mk	11 Jan 2013 20:23:43 -0000	1.1.5.1
+++ build_config/Linux.intel.default/build_rules.mk	10 Sep 2013 11:21:28 -0000
@@ -1,4 +1,4 @@
-# $Id: build_rules.mk,v 1.1.5.1 2013-01-11 20:23:43 mathomp4 Exp $
+# $Id: build_rules.mk,v 1.89.4.1 2012/03/26 23:09:33 theurich Exp $
 #
 # Linux.intel.default
 #
@@ -79,10 +79,16 @@ ESMF_MPIMPMDRUNDEFAULT  = mpiexec $(ESMF
 else
 ifeq ($(ESMF_COMM),openmpi)
 # OpenMPI --------------------------------------------------
-ESMF_CXXCOMPILECPPFLAGS+= -DESMF_NO_SIGUSR2
+ifeq ($(shell $(ESMF_DIR)/scripts/available mpifort),mpifort)
+ESMF_F90DEFAULT         = mpifort
+ESMF_CXXLINKLIBS       += -lmpi_mpifh
+else
 ESMF_F90DEFAULT         = mpif90
-ESMF_CXXDEFAULT         = mpicxx
 ESMF_CXXLINKLIBS       += -lmpi_f77
+endif
+ESMF_CXXCOMPILECPPFLAGS+= -DESMF_NO_SIGUSR2
+ESMF_F90LINKLIBS       += -lmpi_cxx
+ESMF_CXXDEFAULT         = mpicxx
 ESMF_MPIRUNDEFAULT      = mpirun $(ESMF_MPILAUNCHOPTIONS)
 ESMF_MPIMPMDRUNDEFAULT  = mpiexec $(ESMF_MPILAUNCHOPTIONS)
 else
@@ -162,6 +168,7 @@ ESMF_F90COMPILEOPTS += -threads
 ESMF_CXXCOMPILEOPTS += -pthread
 ESMF_F90LINKOPTS    += -threads
 ESMF_CXXLINKOPTS    += -pthread
+ESMF_SL_LIBOPTS     += -pthread
 endif
 
 ############################################################
@@ -218,4 +225,4 @@ ESMF_SO_CXXLINKOPTSEXE  = -Wl,-export-dy
 ############################################################
 # 3rd party code dependency: PIO
 #
-#TODO: commented out for 5.1.0 release: ESMF_PIODEFAULT = internal
+#TODO: activate this once PIO support is stable: ESMF_PIODEFAULT = internal
Index: build_config/Linux.pgi.default/build_rules.mk
===================================================================
RCS file: /cvsroot/baselibs/Baselibs/src/esmf/build_config/Linux.pgi.default/build_rules.mk,v
retrieving revision 1.13
diff -b -u -p -r1.13 build_rules.mk
--- build_config/Linux.pgi.default/build_rules.mk	26 Feb 2013 17:43:20 -0000	1.13
+++ build_config/Linux.pgi.default/build_rules.mk	10 Sep 2013 11:21:28 -0000
@@ -1,4 +1,4 @@
-# $Id: build_rules.mk,v 1.13 2013-02-26 17:43:20 mathomp4 Exp $
+# $Id: build_rules.mk,v 1.61.2.2 2012/03/26 23:09:35 theurich Exp $
 #
 # Linux.pgi.default
 #
@@ -28,9 +28,7 @@ ESMF_MPIRUNDEFAULT      = $(ESMF_DIR)/sr
 else
 ifeq ($(ESMF_COMM),mpi)
 # Vendor MPI -----------------------------------------------
-ESMF_F90DEFAULT         = mpif90
 ESMF_F90LINKLIBS       += -lmpi -lmpi++
-ESMF_CXXDEFAULT         = mpicxx
 ESMF_CXXLINKLIBS       += -lmpi -lmpi++
 ESMF_MPIRUNDEFAULT      = mpiexec_mpt $(ESMF_MPILAUNCHOPTIONS)
 ESMF_MPIMPMDRUNDEFAULT  = mpiexec_mpt $(ESMF_MPILAUNCHOPTIONS)
@@ -83,10 +81,16 @@ ESMF_MPIMPMDRUNDEFAULT  = mpiexec $(ESMF
 else
 ifeq ($(ESMF_COMM),openmpi)
 # OpenMPI --------------------------------------------------
-ESMF_CXXCOMPILECPPFLAGS+= -DESMF_NO_SIGUSR2
+ifeq ($(shell $(ESMF_DIR)/scripts/available mpifort),mpifort)
+ESMF_F90DEFAULT         = mpifort
+ESMF_CXXLINKLIBS       += -lmpi_mpifh
+else
 ESMF_F90DEFAULT         = mpif90
-ESMF_CXXDEFAULT         = mpicxx
 ESMF_CXXLINKLIBS       += -lmpi_f77
+endif
+ESMF_CXXCOMPILECPPFLAGS+= -DESMF_NO_SIGUSR2
+ESMF_F90LINKLIBS       += -lmpi_cxx
+ESMF_CXXDEFAULT         = mpicxx
 ESMF_MPIRUNDEFAULT      = mpirun $(ESMF_MPILAUNCHOPTIONS)
 ESMF_MPIMPMDRUNDEFAULT  = mpiexec $(ESMF_MPILAUNCHOPTIONS)
 else
@@ -227,4 +231,4 @@ ESMF_SO_CXXLINKOPTSEXE  = -Wl,-export-dy
 ############################################################
 # 3rd party code dependency: PIO
 #
-#TODO: commented out for 5.1.0 release: ESMF_PIODEFAULT = internal
+#TODO: activate this once PIO support is stable: ESMF_PIODEFAULT = internal
